src/components/albums/albums.service.js

```diff
@@ -0,0 +1,20 @@
+export class AlbumsService {
+  constructor($q, $http) {
+    'ngInject;'
+    this.$q = $q;
+    this.$http = $http;
+    this.baseUrl = '/search?q=';
+  }
+  
+  getAlbums(albumName) {
+    let finalAlbumName = albumName.replace(/ /g, '+');
+    let queryUrl = this.baseUrl + finalAlbumName;
+    let deferred = this.$q.defer();
```

Comment:
use the new syntax of $q, instead of `this.$q.defer` just do `return $q((resolve, reject) => {})`